src/components/Canvas/Editor.tsx

```diff
@@ -569,6 +586,56 @@ export default class Editor extends EventDispatcher {
     this.interactionLayer.setDpr(dpr);
   }
 
+  /**
+   * @summary Sets the data of the pixel array (use with caution, since data will be overwritten!)
+   * @param data Array of PixelModifyItem (It must be a rectangular array, i.e. all rows must have the same length)
+   */
+  setData(data: Array<Array<PixelModifyItem>>) {
```

Comment:
question) #50 can you elaborate the context? is it to provide more efficient way than colorPixels hook?

Reply:
I did not exactly think of the use cases, but one could use this for many purposes.

1. Choose an image from a gallery
A user(programmer) could want to build a gallery of pixel art, and allow customers to change the dotting data every time a new pixel art is selected. We could just use one Dotting component to handle the change of pixel art

2. Bring data from the server
A user(programmer) would want to initialize the canvas, but maybe there is a time lag when bringing the saved data. The user could call `setData` in those cases.

Different from colorPixels hook, where you have to specify which pixel indices you would like to color, `setData` is a function that allows users(programmers) to change the whole data of the pixel canvas.